pkg/server/handlers.go

```diff
@@ -355,24 +421,23 @@ func updateDeployment(w http.ResponseWriter, r *http.Request) {
 	getDep.Spec.Template.Annotations["publicHosts"] = cacheAnnotations["publicHosts"]
 	getDep.Spec.Template.Annotations["privateHosts"] = cacheAnnotations["privateHosts"]
 
-	if tempJSON.PrivateHosts != nil {
-		getDep.Spec.Template.Annotations["privateHosts"] = *tempJSON.PrivateHosts
-	}
-
-	if tempJSON.PublicHosts != nil {
-		getDep.Spec.Template.Annotations["publicHosts"] = *tempJSON.PublicHosts
-	}
-
-	if apigeeKVM {
-		for index, val := range tempJSON.EnvVars {
-			if val.ValueFrom != (&apigee.ApigeeEnvVarSource{}) {
-				// Gotta go retrieve the value from apigee KVM
-				// In the future we may support other ref types
-				apigeeClient := apigee.Client{Token: r.Header.Get("Authorization")}
-				tempJSON.EnvVars[index], err = apigee.EnvReftoEnv(val.ValueFrom, apigeeClient, pathVars["org"], pathVars["env"])
-				if err != nil {
-					errorMessage := fmt.Sprintf("Failed at EnvReftoEnv: %v\n", err)
-					http.Error(w, errorMessage, http.StatusInternalServerError)
+	for index, val := range tempJSON.EnvVars {
```

Comment:
You are correct that this is identical code. I'll move it into a function and place it in `pkg/server/helper.go`